### PR TITLE
Deploy command times out for processes with a timer start event

### LINKED ISSUE #6911: Deploy command times out for processes with a timer start event
**Describe the bug**
If a process is deployed that contains a timer start event then the deploy command times out (i.e. no response is received).

_Additional observations_ by @menski:
 
What is interesting is that everytime I try to deploy the diagram the following log messages shows up:
```
2021-04-29 13:53:05.355 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-1] INFO 
      org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@68fe7458)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@3ac3fdd, clock: io.zeebe.el.impl.ZeebeFeelEngineClock@2246a945, configuration: Configuration(false)]
```

**To Reproduce**

Deploy a process with a timer start event. 
image

For example:
[timerProcess_v_1.bpmn20.tar.gz](https://github.com/camunda-cloud/zeebe/files/6398358/timerProcess_v_1.bpmn20.tar.gz)

**Environment:**
 - OS: 
 - Zeebe Version: 1.0.0-alpha7
 - Configuration:

### PR DESCRIPTION
**Pull request by saig0**
## Description

* send a response of the deploy command if a process contains a timer start event
* avoid missing the response by adding it explicitly as a side-effect
  * by default, the response is written as a side-effect
  * if the side-effects are overridden by `sideEffects::accept` (as it was on calling `catchEventBehavior.subscribeToTimerEvent()`) then no response is sent implicitly 
  * created follow-up issue to make it harder to fall into this trap: #6919

## Related issues

closes #6911 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/test/java/io/zeebe/engine/processing/deployment/model/validation/SuccessfulDeploymentTest.java [hunk 1]
```diff
@@ -23,17 +41,70 @@
...
-            .deploy();
+    final var deployment = performDeployment.apply(engine.deployment());
 
     // then
     assertThat(deployment.getIntent()).isEqualTo(DeploymentIntent.CREATED);
     assertThat(deployment.getValue().getDeployedProcesses()).hasSize(1);
   }
+
+  @Test
+  public void shouldSendResponse() {
```
Comment (Zelldon) [inline engine/src/test/java/io/zeebe/engine/processing/deployment/model/validation/SuccessfulDeploymentTest.java:89]: Do we want also one for rejection response?

**Review by ChrisKujawa — APPROVED** [review]
Thanks :+1: 

Maybe in general it would be better that we have the queue in the interface such that we can only add stuff and not remove/replace things. (?)

Small question is it still the case that you want to abandon the side effect stuff at some point?

**Comment by saig0** [comment]
> Maybe in general it would be better that we have the queue in the interface such that we can only add stuff and not remove/replace things. (?)

Yes. Let's do something like this in the follow-up issue #6919 :+1: 

> Small question is it still the case that you want to abandon the side effect stuff at some point?

I'm not sure. We could drop the usage for some cases, like message correlation. But we also use it to schedule a timer.

File: engine/src/test/java/io/zeebe/engine/processing/deployment/model/validation/SuccessfulDeploymentTest.java (hunk 1 above)
Comment (saig0) [reply engine/src/test/java/io/zeebe/engine/processing/deployment/model/validation/SuccessfulDeploymentTest.java:89]: In general, it is a good idea to have a test in the engine. However, we have one in the QA module 
`io.zeebe.broker.it.client.command.CreateDeploymentTest#shouldRejectDeployIfProcessIsInvalid`.

**Pull request merged**